app/views/fragments/tagging/googleTagManager.scala.html

```diff
@@ -0,0 +1,26 @@
+@()
+<script>
+  dataLayer = [{
+    'orderId': getOrderId(),
+  }];
+  //Generates a uuid to anonymously identify this user in this session.
+  //We need this to prevent page refreshes on conversion pages being
+  //treated as new conversions
+  function getOrderId() {
+    let orderId = sessionStorage.getItem('orderId');
+    if (orderId === null){
+      orderId = uuid();
+      sessionStorage.setItem('orderId', orderId);
+    }
+    return orderId;
+  }
+  //see: https://gist.github.com/LeverOne/1308368
+  function uuid(){for(b=a='';a++<36;b+=a*51&52?(a^15?8^Math.random()*(a^20?16:4):4).toString(16):'-');return b}
```

Comment:
Can we wrap this in a function scope to avoid bleed into the global scope? Something like this:

```js
(function () {
  window.dataLayer = [{
    'orderId': getOrderId(),
  }];
  //Generates a uuid to anonymously identify this user in this session.
  //We need this to prevent page refreshes on conversion pages being
  //treated as new conversions
  function getOrderId() {
    let orderId = sessionStorage.getItem('orderId');
    if (orderId === null){
      orderId = uuid();
      sessionStorage.setItem('orderId', orderId);
    }
    return orderId;
  }
  //see: https://gist.github.com/LeverOne/1308368
  function uuid(){for(b=a='';a++<36;b+=a*51&52?(a^15?8^Math.random()*(a^20?16:4):4).toString(16):'-');return b}
})();
```